.github/workflows/build-website.yml

```diff
@@ -1,75 +1,91 @@
-
 name: build website
-on:
+on: 
   push:
   pull_request:
     types: [opened, synchronize]
 
+
 jobs:
   build:
     name: website
     runs-on: ubuntu-latest
-    # TODO(strager): Use an image based on
-    # quick-lint-js-web.
-    container: ghcr.io/quick-lint/quick-lint-js-github-builder:v3
+    container: ghcr.io/k1ng440/quick-lint-js-website-builder:v1
     env:
       QLJS_COLLECT_COPYRIGHT_NO_WARNINGS: 1
+      EM_VERSION: 2.0.4
+      EM_CACHE_FOLDER: 'emsdk-cache'
+
     steps:
-      - uses: quick-lint/setup-emsdk@v7
-        with:
-          version: 2.0.4
       - name: checkout
         uses: actions/checkout@v2
 
+      - name: Setup emsdk cache
+        id: cache-system-libraries
+        uses: actions/cache@v2
+        with:
+          path: ${{env.EM_CACHE_FOLDER}}
+          key: ${{env.EM_VERSION}}-${{ runner.os }}
+
+      - uses: quick-lint/setup-emsdk@v7
+        with:
+          version: ${{env.EM_VERSION}}
+          actions-cache-folder: ${{env.EM_CACHE_FOLDER}}
+
       - name: C++ configure
-        run: emcmake cmake -S . -B build -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_C_FLAGS='-Os -flto' -DCMAKE_CXX_FLAGS='-Os -flto'
+        run: emcmake cmake -S . -B build -G Ninja -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=NO -DCMAKE_C_FLAGS='-Os -flto' -DCMAKE_CXX_FLAGS='-Os -flto'
       - name: C++ build
         run: emmake cmake --build build --target quick-lint-js-vscode quick-lint-js-vscode-licenses
       - name: C++ install
         run: emmake cmake --install build --component vscode --prefix website/public/demo
+      
+      - name: get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+      - uses: actions/cache@v1
+        id: yarn-cache
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
 
       - name: JS configure
-        run: cd website && yarn
+        working-directory: website
+        run: yarn
+
       - name: JS test
-        run: cd website && yarn test
+        working-directory: website
+        run: yarn test
 
       - name: build site
-        run: cd website && yarn build
+        working-directory: website
+        run: yarn build
+
       - name: bundle site
-        run: cd website && zip -r ../website.zip www
+        working-directory: website
+        run: zip -r ../website.zip www
 
-      - name: set up check-links
+      - name: check links
         run: |
-          python3.7 -m venv ENV
-          ENV/bin/pip install bs4 requests
-      - name: check links (dev)
-        run: |
-          cd website/
-          yarn start localhost 9001 &
-          sleep 0.5  # HACK(strager): Wait for server to start.
-          cd ../
-          ./ENV/bin/python ./tools/check-links http://localhost:9001/
-          kill %1
-        shell: bash
```

Comment:
Must fix: Why did you delete `check links (dev)`? This step is important to make sure the dev server works. Undelete this step.

Reply:
ok, why do we need to check links using apache2 then?

Reply:
Because the dev server and the build can have different characteristics, since the code isn't 100% identical between the two. We've had bugs in each one.